src/semaphore.c

```diff
@@ -630,6 +680,11 @@ _dispatch_group_wait_slow(dispatch_semaphore_t dsema, dispatch_time_t timeout)
 			ret = sem_wait(&dsema->dsema_sem);
 		} while (ret == -1 && errno == EINTR);
 		DISPATCH_SEMAPHORE_VERIFY_RET(ret);
+#elif USE_FUTEX_SEM
+		do {
+			ret = _dispatch_futex_wait(&dsema->dsema_futex, NULL);
+		} while (ret == -1 && errno == EINTR);
```

Comment:
If you follow the code down to _dispatch_futex_wait_slow
it indicates the resason that <timeout> needs to be recomputed.
If you want I can massage that whole code, but that would require the timeout recomputation
to be brought in here, which seems overdone, instead elevate it back out the caller.
Let me know